src/panels/config/voice-assistants/assist-pipeline-detail/assist-pipeline-detail-stt.ts

```diff
@@ -0,0 +1,102 @@
+import { css, CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property } from "lit/decorators";
+import memoizeOne from "memoize-one";
+import { SchemaUnion } from "../../../../components/ha-form/types";
+import { AssistPipeline } from "../../../../data/assist_pipeline";
+import { HomeAssistant } from "../../../../types";
+
+@customElement("assist-pipeline-detail-stt")
+export class AssistPipelineDetailSTT extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property() public data?: Partial<AssistPipeline>;
+
+  @property() public error?: Record<string, string>;
+
+  private _schema = memoizeOne(
+    (language?: string) =>
+      [
+        {
+          name: "",
+          type: "grid",
+          schema: [
+            {
+              name: "stt_engine",
+              selector: {
+                stt: {
+                  language,
+                },
+              },
+            },
+            {
+              name: "stt_language",
+              selector: {
+                text: {},
+              },
+            },
+          ] as const,
+        },
+      ] as const
+  );
+
+  private _computeLabel = (
+    schema: SchemaUnion<ReturnType<typeof this._schema>>
+  ): string =>
+    this.hass.localize(
+      `ui.panel.config.voice_assistants.assistants.pipeline.detail.form.${schema.name}`
+    );
+
+  protected render() {
+    return html`
+      <div class="section">
+        <div class="intro">
+          <h3>Speech to text</h3>
+          <p>An engine does something that needs some explanation.</p>
```

Comment:
When you are using the pipeline as a voice assistant, the speech-to-text engine turns your voice command into text.